hooks/react-client/useNavigationState.ts

```diff
@@ -0,0 +1,40 @@
+'use client';
+
+import type { RefObject } from 'react';
+import { useContext, useEffect } from 'react';
+
+import { NavigationStateContext } from '@/providers/navigationStateProvider';
+import { debounce } from '@/util/debounce';
+
+const useNavigationState = <T extends HTMLElement>(
+  id: string,
+  ref: RefObject<T>,
+  debounceTime = 300
+) => {
+  const navigationState = useContext(NavigationStateContext);
+
+  const handleScroll = debounce(() => {
+    if (ref.current) {
+      navigationState[id] = {
+        x: ref.current.scrollLeft,
+        y: ref.current.scrollTop,
+      };
+    }
+  }, debounceTime);
+
+  useEffect(() => {
+    const element = ref.current;
+    if (navigationState[id] && navigationState[id].y !== element?.scrollTop) {
+      element?.scroll({ top: navigationState[id].y, behavior: 'instant' });
+    }
+
+    element?.addEventListener('scroll', handleScroll, { passive: true });
+
+    return () => {
+      element?.removeEventListener('scroll', handleScroll);
```

Comment:
No need of brackets here just `return () => ref.current.removeEventListener....`